jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/JdbcConfigSourceFactory.java

```diff
@@ -15,52 +16,100 @@
 import io.smallrye.config.common.MapBackedConfigSource;
 
 public class JdbcConfigSourceFactory implements ConfigSourceFactory {
+
     private static final Logger log = Logger.getLogger(JdbcConfigSourceFactory.class);
     Repository repository = null;
+    JdbcConfigConfig config;
```

Comment:
These should be local variables to the `getConfigSources` method. It prevents the factory to be discarded when is not needed anymore. Only the `ConfigSource` is required.

Reply:
Also, when we are caching the config, the `Repository` is only required to populate the `ConfigSource`. After that, it can be discarded.